### PR TITLE
Force positive size for plaintext output array 

### PR DESCRIPTION
**Pull request by lmctv**
Solves #517. The tests have been verified as failing before applying the enclosed changes to `src/nacl/bindings/crypto_box.py`

### CODE REVIEW

**Comment by lmctv** [comment]
Please send any further comments about the original issue #517 to the proposed solution PR #518

File: src/nacl/bindings/crypto_box.py [hunk 1]
```diff
@@ -304,10 +304,11 @@ def crypto_box_seal_open(ciphertext, pk, sk):
     _clen = len(ciphertext)
     _mlen = _clen - crypto_box_SEALBYTES
```
Comment (lmctv) [inline src/nacl/bindings/crypto_box.py]: I'm unsure about adding a explicit check to raise ValueError when `_mlen < 0`.

**Comment by lmctv** [comment]
@pyca/pynacl-core @reaperhulk I think this is ready for review.

File: src/nacl/bindings/crypto_box.py (hunk 1 above)
Comment (lmctv) [reply src/nacl/bindings/crypto_box.py]: to clarify, if we would decide add a explicit size check, IMO the raised exception should be a `nacl.exception.CryptoError` subclass; both `nacl.exception.TypeError`  and `nacl.exception.ValueError` could make sense.

Comment (reaperhulk) [reply src/nacl/bindings/crypto_box.py]: If we do an explicit check why not just do `_mlen <= 0` and then we don't need to do the `max()` code below. Or is it possible to seal an empty byte string such that when you call crypto_box_seal_open you would expect an empty byte string response?

Comment (lmctv) [reply src/nacl/bindings/crypto_box.py]: While I didn't add a explicit test for that case, I verified sealedbox box can ship **and verify** a zero-length message while preparing the PR. If you think the test can be useful, I can either add it in this PR, or in a follow-up one.

Comment (lmctv) [reply src/nacl/bindings/crypto_box.py]: Still, if we add a explicit check for `_mlen < 0`, we can skip the second `max`.

Comment (reaperhulk) [reply src/nacl/bindings/crypto_box.py]: Let's add the negative test so we can remove that second max
